### PR TITLE
Decode SEF with multiple tag ids

### PR DESCRIPTION
**Pull request by Waldbaer88**
PR for Issue #24980

### Summary of Changes
A link including an array of tag ids like this:
`index.php?option=com_tags&view=tag&id[0]=1&id[1]=2`
will be like the following after SEF-conversion:
`/component/tags/tag/1,2`

Until now, any id after the first of the SEF-link will be truncated by the line I changed - my fix will explode it again to include all of them.

### Testing Instructions
**Part I - See the problem**
1. In a layout override, add a link like this one (make sure the ids are real tag ids on your site):
`<a href="index.php?option=com_tags&view=tag&id[0]=1&id[1]=2">my test link</a>`
2. Make sure SEF is turned off in the global settings. Your link should work as expected and open a site showing _both tags_.
3. Turn on SEF in the global configuration. Reload your test-override. Your link URL will be converted. Test the link.

**Expected result**
Same as with SEF turned off: A site showing _all submitted tags_.

**Actual result**
You will get a site showing _only the first tag_ in the URL but ignore the other one.

### Testing Instructions
**Part II - Fix the problem**
1. Apply this PR.
2. Test your link again - it should now work and show all tags as expected, no matter if SEF is turned on or off.

### Documentation Changes Required
None, no feature changes.

### CODE REVIEW

File: components/com_tags/models/tag.php [hunk 1]
```diff
@@ -178,7 +178,12 @@ protected function populateState($ordering = 'c.core_title', $direction = 'ASC')
 		$this->setState('params', $params);
 
 		// Load state from the request.
-		$ids = ArrayHelper::toInteger($app->input->get('id', array(), 'array'));
+		$ids = $app->input->get('id', array(), 'array');
+		if (count($ids) == 1)
```
Comment (Quy) [inline components/com_tags/models/tag.php:183]:
```suggestion

		if (count($ids) == 1)
```

File: components/com_tags/models/tag.php [hunk 2]
```diff
@@ -178,7 +178,12 @@ protected function populateState($ordering = 'c.core_title', $direction = 'ASC')
...
+		{
+			$ids = explode(',', $ids[0]);
+		}
+		$ids = ArrayHelper::toInteger($ids);
```
Comment (Quy) [inline components/com_tags/models/tag.php:188]:
```suggestion

		$ids = ArrayHelper::toInteger($ids);
```

**Comment by ghost** [comment]
@kofaysi can you please test this PR (as its about Tags)?

**Review by Hackwar — APPROVED** [review]

**Comment by kofaysi** [comment]
I have tested this item :white_check_mark: successfully on 8d777676094795386293ac408a8001cecff6fcbe<br /><br />Works as described. Thank you!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25065">issues.joomla.org/tracker/joomla-cms/25065</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 8d777676094795386293ac408a8001cecff6fcbe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25065">issues.joomla.org/tracker/joomla-cms/25065</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25065">issues.joomla.org/tracker/joomla-cms/25065</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Comment by kofaysi** [comment]
I just checked that listing multiple tags (2, 3, ... pcs) concatenates tags in the title without separator. The result for four multi-word tags may be "tag 1 tag 2 tag named something this is a tag". It makes easier to understand the results, if a separator is used: vertical line | or logical text "OR": "Content for: tag 1 OR tag 2 OR tag named something OR this is a tag". This should go to a new issue to describe and to be solved, right?

**Comment by HLeithner** [comment]
Yes but for j4

**Pull request merged**